pheno_utils/pheno_loader.py

```diff
@@ -544,18 +547,33 @@ def __concat__(self, df1, df2, keep_undefined_research_stage=False):
             self.is_value_in_index(df2, 'undefined', 'research_stage') and not keep_undefined_research_stage:
         
             warnings.warn('filtering "undefined" research_stage..')
-            df = self.join_and_filter_undefined_research_stage(df1, df2)
+            df = self.join_and_filter_undefined_research_stage(df1, df2, how)
             return df
         
-        return df1.join(df2, how='outer')
-        
+        return df1.join(df2, how=how)
+    
+    def merge_all_tables(self) -> pd.DataFrame:
```

Comment:
I'm wondering if there's any reason for this merge to also follow the keep_undefined flag (e.g., use __concat__). What do you think?

Reply:
Good idea. I'll add some parameters to concat.